composer/datasets/in_context_learning_evaluation.py

```diff
@@ -824,6 +827,241 @@ def collate_fn(self, data):
+
+	The input format is expected to be a jsonl file with the following fields:
+	- task_id: label of given task
+	- prompt: the code snippet that must be completed
+    - entry_point: the entry to the function/code snippet to generate
+	- canonical_solution: working solution
+	- test: checker
+	- test_inputs: list of test inputs
+    - test_outputs: list of test outputs
+
+	Args:
+		dataset_uri (str): Either a local path, or a remote path beginning with ``s3://``, or another backend
+            supported by :meth:`composer.utils.maybe_create_object_store_from_uri`. Dataset must consist of rows of JSON data points with "task_id",
+            "prompt", "entry_point", "canonical_solution", "test", "test_inputs", and "test_outputs". See tests/datasets/local_data/human_eval_small.jsonl.
+        tokenizer (Union[transformers.PreTrainedTokenizer, transformers.PreTrainedTokenizerFast]): The tokenizer used to map between strings and token ids
+        batch_size (int): Size of a batch used for eval
+        max_seq_len (int): The maximum sequence length supported by the model
+        pad_tok_id (int): The special token reserved for padding batches
+        num_fewshot (int): The number of complete fewshot examples to prepend before each test example
+        prompt_string (str): Prompt string to put once before all fewshot examples/test examples (e.g. 'translate english to french')
+        example_delimiter (str): Separator that goes between individual (context, answer) pairs (e.g. '\n')
+        destination_path (str): Temporary path to store downloaded datasets
+        code_prelimiter (str): String to put before each code prompt (e.g. 'Q: ')
+        fewshot_random_seed (int): Random seed to use for fewshot sampling
+        num_evals: how many outputs to generate per prompt
+
+	"""
+
+    def __init__(
+        self,
+        dataset_uri: str,
+        tokenizer: Union[transformers.PreTrainedTokenizer, transformers.PreTrainedTokenizerFast],
+        max_seq_len: int,
+        pad_tok_id: int,
+        num_fewshot: int,
+        prompt_string: str,
+        example_delimiter: str,
+        destination_path: str,
+        code_prelimiter: str,
+        fewshot_random_seed: int,
+        num_evals: int,
+    ):
+        try:
+            from datasets import load_dataset  # pyright: ignore [reportGeneralTypeIssues]
+        except ImportError as e:
+            raise MissingConditionalImportError(extra_deps_group='nlp',
+                                                conda_package='datasets',
+                                                conda_channel='conda-forge') from e
+        with dist.local_rank_zero_download_and_wait(destination_path):
+            if dist.get_local_rank() == 0:
+                get_file(dataset_uri, destination_path, overwrite=True)
+        dataset = load_dataset('json', data_files=destination_path, split='train', streaming=False)
+        self.samples = list(
+            dataset.map(
```

Comment:
I recently updated composer ICL to enforce that all white space is stripped from the elements of each jsonl upon loading...this leaves all white space delimiters to be added in by composer. Can you rebase and confirm that the same should apply for this dataset type?

https://github.com/mosaicml/composer/pull/2306/files

Reply:
Discussed offline, shouldn't be stripped.